gio/src/socket.rs

```diff
@@ -27,22 +29,24 @@ impl Socket {
             if error.is_null() {
                 Ok(from_glib_full(ret))
             } else {
-                libc::close(fd);
+                let _ = OwnedFd::from_raw_fd(fd);
                 Err(from_glib_full(error))
             }
         }
     }
     #[cfg(windows)]
     #[cfg_attr(docsrs, doc(cfg(windows)))]
-    #[allow(clippy::missing_safety_doc)]
-    pub unsafe fn from_socket(socket: impl IntoRawSocket) -> Result<Socket, glib::Error> {
+    pub fn from_socket(socket: OwnedSocket) -> Result<Socket, glib::Error> {
         let socket = socket.into_raw_socket();
```

Comment:
Did we implement `Into<OwnedFd>` for Socket?

Reply:
No, we can't really. We can't steal the fd from the socket.